src/table/TableWrapper.jsx

```diff
@@ -64,53 +65,76 @@ export default function TableWrapper(props) {
   }
 
   useEffect(() => {
-    const updateSize = () => setTableWidth(rootElement.clientWidth);
-    window.addEventListener('resize', updateSize);
-    tableSection.current.addEventListener('wheel', (evt) => handleScroll(evt, tableSection));
+    const resizeCallback = () => setTableWidth(rootElement.clientWidth);
+    const scrollCallback = (evt) => handleScroll(evt, tableSectionRef);
+    const focusOutCallback = (evt) => handleFocusoutEvent(evt, shouldRefocus, keyboard.blur);
+
+    window.addEventListener('resize', resizeCallback);
+    tableSectionRef.current.addEventListener('wheel', scrollCallback);
+    tableWrapperRef.current.addEventListener('focusout', focusOutCallback);
     return () => {
-      window.removeEventListener('resize', updateSize);
-      tableSection.current.removeEventListener('wheel', (evt) => handleScroll(evt, tableSection));
+      window.removeEventListener('resize', resizeCallback);
+      tableSectionRef.current.removeEventListener('wheel', scrollCallback);
+      tableWrapperRef.current.removeEventListener('focusout', focusOutCallback);
     };
   }, []);
 
   useEffect(() => {
     handleNavigateTop({
-      tableSection,
+      tableSectionRef,
       focusedCellCoord,
       rootElement,
     });
-  }, [tableSection, focusedCellCoord]);
+  }, [tableSectionRef, focusedCellCoord]);
+
+  useDidUpdateEffect(() => {
```

Comment:
What is the benefit of this useDidUpdateEffect?

Reply:
It doesn't run on first render. if we use `useEffect` it will blur even though nothing is focused